### PR TITLE
fix(DeASDA): Add Support for B3 (untested)

### PR DESCRIPTION
**Pull request by GuiHue**
Added PID config for Delta ASDA B3 drives based on ESI file comparison and PID 6080. I do not have a B3 but expect this to work just fine.
Relates to #248

### CODE REVIEW

File: src/devices/lcec_deasda.c [hunk 1]
```diff
@@ -64,8 +67,9 @@ static const drive_operationmodes_t drive_operationmodes[] = {
 // Note that DeASDA refers to A2-E series of drives and is deliberatly not refering to A2 in its name to ensure compatability with legace
 // configurations.
 static lcec_typelist_t types[] = {
-    {"DeASDA", LCEC_DELTA_VID, 0x10305070, LCEC_DEASDA_PDOS, 0, NULL, lcec_deasda_init, lcec_deasda_modparams, FLAG_A2},
-    {"DeASDA3", LCEC_DELTA_VID, 0x00006010, LCEC_DEASDA_PDOS, 0, NULL, lcec_deasda_init, lcec_deasda_modparams, FLAG_A3},
+    {"DeASDA", LCEC_DELTA_VID, 0x10305070, LCEC_DEASDA_PDOS, 0, NULL, lcec_deasda_init, lcec_deasda_modparams, FLAG_LOWRES_ENC, FLAG_SERVO_X2},
+    {"DeASDA3", LCEC_DELTA_VID, 0x00006010, LCEC_DEASDA_PDOS, 0, NULL, lcec_deasda_init, lcec_deasda_modparams, FLAG_HIGHRES_ENC,FLAG_SERVO_X3},
```
Comment (scottlaird) [inline src/devices/lcec_deasda.c:71]: It should be FLAG_HIGHRES_ENC | FLAG_SERVO_X3 here(and neighboring lines)

**Review by scottlaird — APPROVED** [review]

File: src/devices/lcec_deasda.c (hunk 1 above)
Comment (GuiHue) [reply src/devices/lcec_deasda.c]: right, taken care off (actually by removing the second set of flags - not required atm

**Comment by GuiHue** [comment]
@scottlaird I spent some more time on this, adding pins for torque readout and status of homing, neg/pos-lim and DI_x pins.
works fine on A2 and A3 and therefore likely also for B3

File: documentation/deasda.md [hunk 2]
```diff
@@ -55,71 +56,86 @@ TODO()
...
-bit   OUT    lcec.0.A3.extenc-on-home-neg
-bit   OUT    lcec.0.A3.extenc-on-home-pos
-float OUT    lcec.0.A3.extenc-pos
-float OUT    lcec.0.A3.extenc-pos-abs | use for position feedback on axis if linear scale is connected
-float OUT    lcec.0.A3.extenc-pos-enc
-bit   IN     lcec.0.A3.extenc-pos-reset
-s32   OUT    lcec.0.A3.extenc-raw
-u32   OUT    lcec.0.A3.extenc-ref-hi
-u32   OUT    lcec.0.A3.extenc-ref-lo
+37    I/O  lcec.0.A3.enc-ext-hi
```
Comment (scottlaird) [inline documentation/deasda.md:63]: 37?

File: src/devices/lcec_deasda.c [hunk 3]
```diff
@@ -148,6 +165,18 @@ static const lcec_pindesc_t slave_pins[] = {{HAL_FLOAT, HAL_OUT, offsetof(lcec_d
     {HAL_BIT, HAL_IN, offsetof(lcec_deasda_data_t, fault_reset), "%s.%s.%s.srv-fault-reset"},
     {HAL_BIT, HAL_IN, offsetof(lcec_deasda_data_t, halt), "%s.%s.%s.srv-halt"},
     {HAL_U32, HAL_OUT, offsetof(lcec_deasda_data_t, operation_mode), "%s.%s.%s.srv-operation-mode"},
+    {HAL_FLOAT, HAL_OUT, offsetof(lcec_deasda_data_t, torque), "%s.%s.%s.srv-torque-rel"}, // relative value (5) - hence current would be redundant
+    {HAL_BIT, HAL_OUT, offsetof(lcec_deasda_data_t, di_1), "%s.%s.%s.srv-di1"},
```
Comment (scottlaird) [inline src/devices/lcec_deasda.c:169]: I'd love to see din/dout pins use somewhat standard names.  Right now, here are the flavors being used across the tree:

- `din-X`: digital in/out boards in general, EpoCAT
- `srv-dinX`: EL7041, EL70x1
- `srv-X-dinX`: EL7342
- `din-<NAME>`: OMRG5

Personally, I'd prefer `din-X` for pins where numeric IDs make sense, or `din-<NAME>` where the pins have specific names on the device or in the documentation.  OMRG5 has `din-pot`, `din-pc`, `din-ext1`, `din-ext2`, `din-ext3`, `din-mon0`, `din-mon1`, `din-mon2`, `din-pcl`, `din-stop`, `din-bkir`, and a couple others.

Would `din-X` work for you here?

File: src/devices/lcec_deasda.c [hunk 4]
```diff
@@ -148,6 +165,18 @@ static const lcec_pindesc_t slave_pins[] = {{HAL_FLOAT, HAL_OUT, offsetof(lcec_d
...
+    {HAL_BIT, HAL_OUT, offsetof(lcec_deasda_data_t, di_2), "%s.%s.%s.srv-di2"},
+    {HAL_BIT, HAL_OUT, offsetof(lcec_deasda_data_t, di_3), "%s.%s.%s.srv-di3"},
+    {HAL_BIT, HAL_OUT, offsetof(lcec_deasda_data_t, di_4), "%s.%s.%s.srv-di4"},
+    {HAL_BIT, HAL_OUT, offsetof(lcec_deasda_data_t, di_5), "%s.%s.%s.srv-di5"},
+    {HAL_BIT, HAL_OUT, offsetof(lcec_deasda_data_t, di_6), "%s.%s.%s.srv-di6"},
+    {HAL_BIT, HAL_OUT, offsetof(lcec_deasda_data_t, di_7), "%s.%s.%s.srv-di7"},
+    {HAL_BIT, HAL_OUT, offsetof(lcec_deasda_data_t, neg_lim_switch), "%s.%s.%s.srv-neg-lim"},
```
Comment (scottlaird) [inline src/devices/lcec_deasda.c:176]: following the OMGA example, maybe `din-neg-lim`, `din-pos-lim`, and `din-home`?

**Review by scottlaird — APPROVED** [review]
A couple naming nits, but nothing big.

File: src/devices/lcec_deasda.c (hunk 3 above)
Comment (GuiHue) [reply src/devices/lcec_deasda.c]: Yeah, makes sense. I like din-x.
However, I'd suggest keeping this nomenclature to actual physical inputs (i.e. connected to the servo drive in explicit 1:1 wiring).
This means that all the other pins remain in their current naming state.

File: documentation/deasda.md (hunk 2 above)
Comment (GuiHue) [reply documentation/deasda.md]: Updated.

File: src/devices/lcec_deasda.c (hunk 4 above)
Comment (GuiHue) [reply src/devices/lcec_deasda.c]: done

**Comment by GuiHue** [comment]
@scottlaird Updated and tested.

**Review by scottlaird — APPROVED** [review]

**Comment by scottlaird** [comment]
Thanks!

**Pull request merged**